### DISCUSSION TITLE
External Love

### DISCUSSION THREAD

**Discussion by thedinz** [immich-app/immich#6730] [General]
Lots of talk about this, i know, nothing new.

Also, not complaints, mainly just pleading my case. Importing just doesnt work for my use case, i really want immich to simply monitor a directory, and show me photos in there, if a photo is removed, stop showing it.

Currently, as I'm sure you are aware, my workflow seems to consist of manually checking for new images, and manually running remove offline images.

With all of us home lab users trying to automate as much as possible this is just too much work. I know i know, what a problem to have lol.

But, despite how it may sound, I'm not complaining, just what seems (from a non-dev) a simple feature that i need before i can use immich, for now, im using photoview which does just what i said i need. I like immich much better, its requires my input too often to show me the current live photo structure.

Importing is not an option for those wondering because then i now have 2 copies of my photos, i have one app that downloads them and deletes them as needed, immich would then COPY these to its own folder and now i have 2 copies.

**Comment by alextran1502** [comment]
You can set the auto-scan to any interval that you want. You can find this in Administration > Settings page

![image](https://github.com/immich-app/immich/assets/27055614/40390a38-ef5c-4562-97c8-b08ce0aa8095)

Auto removal is not implemented because you are mounting your file system into the Docker's volume, what if your mount has some problem and the Docker container cannot find the mount so it proceeds to auto-remove all of the entries in the database?

There are currently more work to help with managing the external library. The one that is almost done is the mechanism to auto watch the library folder, which will help with some task you are describing.

https://github.com/immich-app/immich/pull/6192

**Reply by thedinz** [reply]
Wow what a swift reply thank you.

You are correct about the mount, personally, that's why i set mine up so the container wont start if the mount is offline but that dosnt work for evwryone i know.

Ill tell you what even if the "remove offline photos" button was easier to access that would change a lot, like when i click on my username in the top right and would go to account settings if right there i had some options like "New Scan" and "Remove Offline Files" that would make my life much easier.

Just thinking outside the box here.

**Reply by bo0tzz** [reply]
You could set up a cronjob to regularly call the [removeOfflineFiles endpoint](https://immich.app/docs/api/remove-offline-files). At your own peril, of course.

**Reply by thedinz** [reply]
I would totally do this, though not totally sure what the full command would look like?

**Comment by etnoy** [comment]
The latest release has the option to auto watch for filesystem changes

**Reply by thedinz** [reply]
enabled with polling as its an external share, where will it start? No indication its doing anything. photos not updated.

**Reply by etnoy** [reply]
It will only pull in changes. Try adding a new photo to the folder.

**Reply by thedinz** [reply]
<img width="1336" alt="Screen Shot 2024-02-01 at 9 39 03 AM" src="https://github.com/immich-app/immich/assets/68015411/fa660213-9519-44a4-99cd-c3a1a1d13fb0">
Even if the photos already do not exist?
So you can see this SpongeBob one has the exclamation mark, it's no longer on the file system yet, and has not been removed. Your saying I need to add a new photo for it to remove this one? Just confusing, the system clearly knows its not present, so, why not remove it?

**Reply by etnoy** [reply]
Please read the library documentation for how offline files work

**Reply by alextran1502** [reply]
@thedinz There is no auto removal mechanism, because the scenario below can happen

- You edit your `fstab` entries and somehow break the entry to auto-mount your network drive
- Immich cannot see the volume mounted as you specified since the mount is not correctly mounted

With auto removal, all of the entries will be removed from your database, and you will have to reingest/re-processing your entire collection after fixing the mount.

I hope that clears thing up

**Reply by thedinz** [reply]
Perhaps then, i misunderstood what this new feature was. I thought by watching the file system it would pull in additions and remove deletions, so if that is not the case then it's my fault for not understanding.

I did just a new photo, it showed up fast, but still, the deleted photos still exist. This is the #1 feature I think us external library users are looking for. But if it's not feasible then I understand :)

What you are saying about the mount being offline makes sense, as a non-developer its easy for me to say there must be a way around it, I don't know what that way is but... I would think hosting things on a NAS is not uncommon.

I do not know, but i am here and willing to test new features related to this if you do continue to work on it.

**Reply by etnoy** [reply]
As we've said several times now, we don't auto remove assets, you need to run the remove offline files job, as stated clearly in the docs

**Reply by thedinz** [reply]
Perhaps you missed the part where I said it was my fault for misunderstanding the new feature. Perhaps you even further missed how I said (IF) you were to continue to look into it I would help test.

**Reply by alextran1502** [reply]
A series of unfortunate misunderstandings. 

Anyhow, I don't think we will implement the auto-removal feature.

**Reply by thedinz** [reply]
thanks, Alex, fantastic app, either way, users tend to think something is "no good" because what "they" want is not possible.

Not the case here, immich is quite amazing, I'm just going to have to make adjustments on my end to accommodate.

Keep up the good work gents!

**Reply by alextran1502** [reply]
That is OK; we have the responsibility to guard users and shoot their own feet as well, since the adminstration skills varies in the self-hosting space
